src/main/java/application/FormplayerAuthFilter.java

```diff
@@ -48,21 +47,37 @@
     FormSessionRepo formSessionRepo;
 
     @Value("${commcarehq.formplayerAuthKey}")
-    private String authKey;
+    private String formplayerAuthKey;
 
     @Override
     protected void doFilterInternal(HttpServletRequest req, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
-        FormplayerHttpRequest request = new FormplayerHttpRequest((HttpServletRequest) req);
+        FormplayerHttpRequest request = new FormplayerHttpRequest(req);
         if (isAuthorizationRequired(request)) {
             // These are order dependent
             if (getSessionId(request) == null) {
                 setResponseUnauthorized(response, "Invalid session id");
                 return;
             }
 
-            if (authKey != null && authKey.equals(getSessionId(request))) {
-                logger.info("Logging in as touchforms_user");
-                setSmsRequestDetails(request);
+            if (request.getHeader(Constants.HMAC_HEADER) != null) {
```

Comment:
can we also check that the formplayer auth key being used actually has a value (not null or empty string)?